test/integration/targets/user/tasks/main.yml

```diff
@@ -1135,3 +1135,34 @@
   when: ansible_facts.system in ['Linux']
   tags:
     - user_test_local_mode
+
+## check umask
+- block:
+    - name: Create user with 000 umask
+      user:
+        name: umaskuser_test_1
+        umask: "000"
+      register: umaskuser_test_1
+
+    - name: Create user with 077 umask
+      user:
+        name: umaskuser_test_2
+        umask: "077"
+      register: umaskuser_test_2
+
+    - name: check permissions on created home folder
+      stat:
+        path: "{{ user_home_prefix[ansible_facts.system] }}/umaskuser_test_1"
+      register: umaskuser_test_1_path
+
+    - name: check permissions on created home folder
+      stat:
+        path: "{{ user_home_prefix[ansible_facts.system] }}/umaskuser_test_2"
+      register: umaskuser_test_2_path
+
+    - name: Ensure correct umask has been set on created users
+      assert:
+        that:
+          - umaskuser_test_1_path.stat.mode == "0777"
+          - umaskuser_test_2_path.stat.mode == "0700"
+  when: ansible_facts.system == 'Linux' and not (ansible_distribution == 'Fedora' and ansible_distribution_major_version is version('32', '=='))
```

Comment:
Setting `UMASK=000` doesn't work on Fedora 32 (it sets permission to `700` instead of `777`), but I suppose it has nothing to do with this PR, therefore I have ignored Fedora 32 and this test will not be run it.
More info: https://app.shippable.com/github/ansible/ansible/runs/173128/27/tests

Reply:
Maybe I should check permissions of home folder after creation and give a warning if it has not been set correctly.

Is it good?

Reply:
I know that Ansible is a RedHat product too, but I want to criticize Fedora on this behaviour (I use arch btw :slightly_smiling_face: )

The current behaviour is because of `HOME_MODE` directive on `/etc/login.defs` in Fedora 32 which has been set `0700` by default. The shadow package by default has commented this https://github.com/shadow-maint/shadow/blob/master/etc/login.defs#L207, but Fedora changes this and uncommented `HOME_MODE` therefore by default `umask` doesn't do anything.

So I think ignoring Fedora 32 (and later versions to come) is the proper way to test this PR.